File: front_web/src/utils/campanha-imagem.ts
const TOTAL_IMAGENS = 7;

export function imagemCampanha(id: string | number, cdImagem?: string): string {
  if (cdImagem && cdImagem !== "" && cdImagem !== "null" && cdImagem !== "undefined") {
    return `/assets/campanhas/${cdImagem}`;
  }
  const index = (Math.abs(Number(String(id).replace(/\D/g, "").slice(-6) || 1)) % TOTAL_IMAGENS) + 1;
  return `/assets/campanhas/${index}.png`;
}

File: front_mobile/lib/utils/campanha_imagem.dart
const int _totalImagens = 7;

/// Retorna o asset path (1–7) escolhido deterministicamente pelo ID.
/// Imagens enviadas ao servidor não são bundled no app, por isso ignoramos
/// cdImagem e sempre usamos os banners locais.
String imagemCampanhaAsset(String id, {String? cdImagem}) {
  final digits = id.replaceAll(RegExp(r'\D'), '');
  final seed = int.tryParse(
        digits.length > 6 ? digits.substring(digits.length - 6) : digits,
      ) ??
      1;
  final index = (seed.abs() % _totalImagens) + 1;
  return 'assets/campanhas/$index.png';
}
